packages/amplication-client/src/Workspaces/hooks/useWorkspaceSelector.ts

```diff
@@ -54,14 +54,17 @@ const useWorkspaceSelector = (authenticated: boolean) => {
 
   useEffect(() => {
     if (loading && !authenticated) return;
-
     data &&
       data.currentWorkspace.id !== workspace &&
-      history.push(`/${data.currentWorkspace.id}`);
+      history.push({
+        pathname: `/${data.currentWorkspace.id}`,
+        search: location.search,
+      });
 
     data &&
       data.currentWorkspace.id === workspace &&
       setCurrentWorkspace(data.currentWorkspace);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What's the reason we need this? I'd rather avoid these small hacks because they tend to pile up. Can we fix it?